contrib/platform/src/com/sun/jna/platform/win32/Cfgmgr32Util.java

```diff
@@ -96,4 +95,91 @@ public static String CM_Get_Device_ID(int devInst) throws Cfgmgr32Exception {
             return buffer.getWideString(0);
         }
     }
+
+    /**
+     * Utility method to call Cfgmgr32's CM_Get_DevNode_Registry_Property that
+     * allocates the required memory for the Buffer parameter, and returns values of
+     * the appropriate type.
+     *
+     * @param devInst
+     *            Caller-supplied device instance handle that is bound to the local
+     *            machine.
+     * @param ulProperty
+     *            A {@code CM_DRP_}-prefixed constant value that identifies the
+     *            device property to be obtained from the registry. These constants
+     *            are defined in Cfgmgr32.h.
+     * @return An {@link Object} containing the specified registry property for the
+     *         device.
+     *         <p>
+     *         If the property is of type {@link WinNT.REG_SZ}, a
+     *         {@link java.lang.String} is returned.
+     *         <p>
+     *         If the property is of type {@link WinNT.REG_MULTI_SZ}, an array of
+     *         {@link java.lang.String} is returned.
+     *         <p>
+     *         If the property is of type {@link WinNT.REG_DWORD}, an
+     *         {@link java.lang.Integer} is returned.
+     *         <p>
+     *         If the property is of type {@link WinNT.REG_BINARY}, an array of
+     *         {@link java.lang.Byte} is returned.
+     *         <p>
+     *         If no value exists for this property, or an error is encountered,
+     *         returns {@code null}.
+     * @throws Cfgmgr32Exception
+     */
+    public static Object CM_Get_DevNode_Registry_Property(int devInst, int ulProperty) throws Cfgmgr32Exception {
+
+        // Get byte count and type
+        IntByReference size = new IntByReference();
+        IntByReference type = new IntByReference();
+        int ret = Cfgmgr32.INSTANCE.CM_Get_DevNode_Registry_Property(devInst, ulProperty, type, null, size, 0);
+        // If successful in retrieving type and size, should fail with CR_BUFFER_SMALL
+        // Other errors such as CR_NO_SUCH_VALUE should return null
+        if (ret != Cfgmgr32.CR_BUFFER_SMALL) {
```

Comment:
Wouldn't it make sense to raise an `Cfgmgr32Exception` if anything but `CR_SUCCESS`, `CR_BUFFER_SMALL` or `CR_NO_SUCH_VALUE` is returned? I would expect an exception if an error is returned and I would be irritated, if it is swallowed.

Reply:
Yeah, I kept having that thought and going back and forth with how to handle it. I agree null for no value and exception for other error is appropriate.